controller/cmd/service-mirror/cluster_watcher_mirroring_test.go

```diff
@@ -262,6 +264,12 @@ func TestRemoteServiceUpdatedMirroring(t *testing.T) {
 }
 
 func TestRemoteGatewayUpdatedMirroring(t *testing.T) {
+
+	linkerdIP, err := net.ResolveIPAddr("ip", "linkerd.io")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This definitely isn't a blocker -- but this test requires internet connectivity, which is generally something I think we want to avoid.

I think we could avoid this if we hid https://github.com/linkerd/linkerd2/pull/4588/files#diff-d826c2a036f7464e72c567626b5adce9R1150 behind adding a `resolve` strategy onto `RemoteClusterServiceWatcher`.  By default we use an impl that actually hits the network; but in tests we could mock it.

Reply:
Or, could we use `localhost` in the test?